crates/eth_compatibility/src/lib.rs

```diff
@@ -0,0 +1,142 @@
+
+    #[inline]
+    fn deref(&self) -> &Self::Target {
+        &self.0
+    }
+}
+
+impl core::ops::DerefMut for EthereumAddress {
+    #[inline]
+    fn deref_mut(&mut self) -> &mut Self::Target {
+        &mut self.0
+    }
+}
+
+impl ECDSAPublicKey {
+    /// Returns Ethereum address from the ECDSA compressed public key.
+    ///
+    /// # Example
+    ///
+    /// ```
+    /// use ink_eth_compatibility::ECDSAPublicKey;
+    /// let pub_key: ECDSAPublicKey = ECDSAPublicKey {
+    ///     0: [
+    ///         2, 121, 190, 102, 126, 249, 220, 187, 172, 85, 160,  98, 149, 206, 135, 11,
+    ///         7,   2, 155, 252, 219,  45, 206,  40, 217, 89, 242, 129,  91,  22, 248, 23,
+    ///         152,
+    ///     ]
+    /// };
+    ///
+    /// const EXPECTED_ETH_ADDRESS: [u8; 20] = [
+    ///     126, 95, 69, 82, 9, 26, 105, 18, 93, 93, 252, 183, 184, 194, 101, 144, 41, 57, 91, 223
+    /// ];
+    ///
+    /// assert_eq!(*pub_key.to_eth_address(), EXPECTED_ETH_ADDRESS);
+    /// ```
+    pub fn to_eth_address(&self) -> EthereumAddress {
+        use ink_env::hash;
+        use secp256k1::PublicKey;
+
+        // Transform compressed public key into uncompressed.
+        let pub_key = PublicKey::parse_compressed(&self.0)
+            .expect("Unable to parse the compressed ECDSA public key");
+        let uncompressed = pub_key.serialize();
+
+        // Hash the uncompressed public key by Keccak256 algorithm.
+        let mut hash = <hash::Keccak256 as hash::HashOutput>::Type::default();
+        // The first byte indicates that the public key is uncompressed.
+        // Let's skip it for hashing the public key directly.
+        ink_env::hash_bytes::<hash::Keccak256>(&uncompressed[1..], &mut hash);
+
+        // Take the last 20 bytes as an Address
+        let mut result = EthereumAddress::default();
+        result.copy_from_slice(&hash[12..]);
+
+        result
+    }
+
+    /// Returns the `[u8; 32]` from the ECDSA compressed public key.
+    /// It hashes the compressed public key with the blake2b256 algorithm like in substrate.
+    ///
+    /// # Note
+    ///
+    /// `[u8; 32]` can be converted into default `AccountId` with `into()` method.
+    ///
+    /// # Example
+    ///
+    /// ```
+    /// use ink_eth_compatibility::ECDSAPublicKey;
+    /// let pub_key: ECDSAPublicKey = ECDSAPublicKey {
+    ///     0: [
+    ///         2, 121, 190, 102, 126, 249, 220, 187, 172, 85, 160,  98, 149, 206, 135, 11,
+    ///         7,   2, 155, 252, 219,  45, 206,  40, 217, 89, 242, 129,  91,  22, 248, 23,
+    ///         152,
+    ///     ]
+    /// };
+    ///
+    /// const EXPECTED_ACCOUNT_ID: [u8; 32] = [
+    ///     41, 117, 241, 210, 139, 146, 182, 232,  68, 153, 184, 59,   7, 151, 239, 82,
+    ///     53,  85,  62, 235, 126, 218, 160, 206, 162,  67, 193, 18, 140,  47, 231, 55,
+    /// ];
+    ///
+    /// assert_eq!(pub_key.to_account_id(), EXPECTED_ACCOUNT_ID);
+    pub fn to_account_id(&self) -> [u8; 32] {
```

Comment:
```suggestion
    pub fn to_blake2x256(&self) -> [u8; 32] {
```

Reply:
How would you like this? I think we should not over-abstract of what a function does and here it actually just takes those 20 bytes of the Ethereum address and converts it to `[u8; 32]` using the Blake2x256 hashing algorithm. What the user does with it (e.g. converting it to a default account ID) is not really important with respect to this API.

Documentation should cover this particular use case which you are already doing.

Reply:
Only one reason is why this method exists - describe to the developer the way how to generate substrate's account id from ECDSA public key(33 bytes). Removing this information means that we only take the hash from some bytes... 

So if you prefer to not mention the substrate and default account id here, then better to remove this method at all.

But I think that better to keep this method for developers with a description that this method implements the conversion way into `AccountId` from the substrate.

Reply:
I don't know about the exact use cases but you are free to name it `to_default_account_id` and return just that (what you did in the beginning). However, `to_account_id` would just cause confusion with respect to other environment types.

Reply:
Okey, then let's use `to_default_account_id ` with `DefaultEnvironment::AccountId`=)